eth/stagedsync/stage_hashstate.go

```diff
@@ -169,34 +150,120 @@ func PromoteHashedStateCleanly(logPrefix string, db kv.RwTx, cfg HashStateCfg, q
 	)
 }
 
-func keyTransformExtractFunc(transformKey func([]byte) ([]byte, error)) etl.ExtractFunc {
-	return func(k, v []byte, next etl.ExtractNextFunc) error {
-		newK, err := transformKey(k)
+func readPlainStateOnce(
+	logPrefix string,
+	db kv.RwTx,
+	tmpdir string,
+	loadFunc etl.LoadFunc,
+	quit <-chan struct{},
+) error {
+	bufferSize := etl.BufferOptimalSize
+	buffer := etl.NewSortableBuffer(bufferSize)
+
+	accCollector := etl.NewCollector(logPrefix, tmpdir, buffer)
```

Comment:
oops, here is the problem: each collector must have own buffer - for example because at the end of collector.Load collector does dispose buffer.

Reply:
something like 
```
accCollector = etl.NewCollector(logPrefix, tmpdir, etl.NewSortableBuffer(etl.BufferOptimalSize))
storageCollector = etl.NewCollector(logPrefix, tmpdir, etl.NewSortableBuffer(etl.BufferOptimalSize))
```
is ok

Reply:
Another reason: imagine you have 100Gb of accounts and 400Gb of storage. Then it will store many files inside `etl-tmp` folder. And collector.Load will read all this file (in sequential manner - and do merge-sort). If 2 collectors will use 1 buffer, then you will read 500Gb of `etl-tmp` files twice from disk.